### PR TITLE
Fix Javascript issue on non sqlite databases

### LINKED ISSUE #3166: Some notices about vaultwarden/admin dashboard
### Subject of the issue

Updating vaultwarden/admin dashboard notes 

### Deployment environment

* vaultwarden version:  1.27.0-9366e314

* Install method: Docker

* Clients used: vaultwarden admin dashboard

* Reverse proxy and version: Apache/2.4.6

* MySQL/MariaDB or PostgreSQL version: 10.6.4-MariaDB-1:10.6.4+maria~focal

### Steps to reproduce
- dev-tools console reports an error about adminsetting.js
admin-settings
-  @**Settings-->SMTP Email Settings-->Test SMTP**:  if you try to send mail by clicking _Send Test email_ button it doesn't work. The mail is correctly sent if you push _enter_ when the email field is focused.
- On the first access to the admin dashboard, clicking on _Diagnostics_, browser showed a **Bitwarden  error page** (it's my fault, i neither took a screenshot or note down URL about). It only happened once.

### Your environment (Generated via diagnostics page)
* Vaultwarden version: 1.27.0-9366e314
* Web-vault version: v2023.1.1
* OS/Arch: linux/x86_64
* Running within Docker: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: MySQL
* Database version: 10.6.4-MariaDB-1:10.6.4+maria~focal
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 100,
  "database_timeout": 30,
  "database_url": "*****://**********************************************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://********************",
  "domain_origin": "*****://********************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "password-dev",
  "invitations_allowed": false,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Debug",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": true,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "************",
  "signups_verify": true,
  "signups_verify_resend_limit": 0,
  "signups_verify_resend_time": 60,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "*********************",
  "smtp_from_name": "********************.it",
  "smtp_host": "******************",
  "smtp_password": null,
  "smtp_port": 25,
  "smtp_security": "off",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

### PR DESCRIPTION
**Pull request by BlackDex**
When a non sqlite database is used, loading the admin interface fails
because the backup button is not generated.
This PR is solves it by checking if the elements are valid.

Also made some other js/css changes and fixed some eslint errors.
Showing `_post` errors is better now.

Update jquery to latest version.

Fixes #3166

### CODE REVIEW

File: src/static/scripts/admin_diagnostics.js [hunk 1]
```diff
@@ -212,8 +212,15 @@ document.addEventListener("DOMContentLoaded", (/*event*/) => {
...
-        generateSupportString(diag_json);
-    });
-    document.getElementById("copy-support").addEventListener("click", copyToClipboard);
+    const btnGenSupport = document.getElementById("gen-support");
+    if (btnGenSupport) {
+        btnGenSupport.addEventListener("click", () => {
+            generateSupportString(diag_json);
+        });
+    }
+    document.getElementById("gen-support")
```
Comment (stefan0xC) [inline src/static/scripts/admin_diagnostics.js:221]: I don't think this second `document.getElementById("gen-support")` necessary?

File: src/static/scripts/admin_settings.js [hunk 2]
```diff
@@ -148,7 +148,9 @@ function masterCheck(check_id, inputs_query) {
     const checkbox = document.getElementById(check_id);
     const onChange = onChanged(checkbox, inputs_query);
     onChange(); // Trigger the event initially
-    checkbox.addEventListener("change", onChange);
+    if (checkbox) {
```
Comment (stefan0xC) [inline src/static/scripts/admin_settings.js:151]: Not sure if this can fail but I think the `if (checkbox) {` should be before the call on `onChanged` ?

**Review by stefan0xC — CHANGES_REQUESTED** [review]

File: src/static/scripts/admin_settings.js (hunk 2 above)
Comment (BlackDex) [reply src/static/scripts/admin_settings.js:151]: It shouldn't fail but it also doesn't harm to move indeed.

File: src/static/scripts/admin_diagnostics.js (hunk 1 above)
Comment (BlackDex) [reply src/static/scripts/admin_diagnostics.js:221]: That's what you get after a long day of working and no much sleep haha.

Comment (BlackDex) [reply src/static/scripts/admin_diagnostics.js:221]: and there are some more issues i think. Need to do some more testing on this :).
Thanks for checking.

Comment (BlackDex) [reply src/static/scripts/admin_diagnostics.js]: Made it more robust now.

File: src/static/scripts/admin_settings.js (hunk 2 above)
Comment (BlackDex) [reply src/static/scripts/admin_settings.js]: Same as above, i made it more robust now.

**Review by dani-garcia — APPROVED** [review]

**Pull request merged**